integration/spark-common/src/main/scala/org/apache/carbondata/api/CarbonStore.scala

```diff
@@ -120,9 +120,16 @@ object CarbonStore {
               (-1L, -1L)
             }
           } else {
-            // for batch segment, we can get the data size from table status file directly
-            (if (load.getDataSize == null) -1L else load.getDataSize.toLong,
-              if (load.getIndexSize == null) -1L else load.getIndexSize.toLong)
+            // If the added segment is other than carbon segment then we can only display the
+            // data size and not index size and for batch segment, we can get the data size
+            // from table status file directly
+            if (!load.getFileFormat.equals(FileFormat.COLUMNAR_V3) &&
```

Comment:
please add a utility function in FileFormat to check whether it is carbon format